### DISCUSSION TITLE
Dynamically FIELD by selecting a parameter in variables

### DISCUSSION THREAD

**Discussion by sebas1986** [grafana/grafana-json-datasource#220] [Q&A]
Hi, first of all thanks for sharing this project. I find it very useful.
My question is related to handling variable values within the Fields definition, with JSONata parser.
In particular, I would like to create a field as follows:

**$.doc.history.$myVariable**

where $myVariable is obviously a grafana dashboard variable.
As you can imagine it does not work, but the interesting bit is that it does note go in error, while instead it still polulates the table column but with all the rows with inside the value of the variable. 
So for instance if the variable is "username", i expect the parser to work as:

**$.doc.history.username**

but actually, it creates a column where I find "username" in all the rows (which is not what I expect).
Any suggestion/idea?

Many thanks,

Sebastian

**Comment by bode689** [comment]
Same for me. It populates my result with the text of the variable, not making it assume the correct sense as per JSONata syntax.
I've not found any workaround on this yet unfortunately. Tried any combination of brackets/quotes etc.

Matteo
